SparkPost/matchbox
# UX-730 Do not pass system props to DOM on Text and Box

### What Changed
- Adds `clean` to `Box` and `Text`
- Replaces Cypress tests that relied on system props in the DOM
- Fixes weird usage of the `Error` component that was breaking 

### How To Test or Verify
- Run libby and visit Box or Text stories
- Verify system props are not passed to the DOM

### PR Checklist
